src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/FootnoteServiceTests.cs

```diff
@@ -1205,13 +1916,110 @@ await using (var statisticsDbContext = InMemoryStatisticsDbContext(contextId))
             }
         }
 
+        private static async Task SeedDatabase(string contextId, Release release, IReadOnlyList<Subject>? subjects = null, IReadOnlyList<ReleaseSubject> releaseSubjects = null, IReadOnlyList<Footnote>? footnotes = null)
+        {
+            await using (var contentDbContext = InMemoryApplicationDbContext(contextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(contextId))
+            {
+                await statisticsDbContext.Release.AddAsync(release);
+
+                if (!subjects.IsNullOrEmpty())
+                {
+                    await statisticsDbContext.Subject.AddRangeAsync(subjects!);
+                }
```

Comment:
I think you could argue that saving the subjects here is a bit pointless as they will be saved by saving the releasesubjects, and then it's one less parameter to pass in.

However on reflection I suppose this allows for saving a subject and associated filters/indicators without necessarily linking it to a releasesubject, and it's probably better to be explicit about the saving anway. 🙂

Reply:
I agree, but I also thought it would be best to be more explicit in case you did not attach the subject to the releaseSubject